examples_v10x/systick_irq/systick_irq.c

```diff
@@ -0,0 +1,139 @@
+/*
+* Example for using SysTick with IRQs
+* 03-25-2023 E. Brombaugh
+* 05-12-2023 C. Lohr (Modified to reflect updated sysclk)
+* 09-25-2024 ADBeta (Minor updates to main loop, comments and added
+*                    convenient macro function)
+* 29-06-2025 monte-monte (Adapted to CH32V103)
+*/
+
+#include "ch32fun.h"
+#include <stdio.h>
+
+// Number of ticks elapsed per millisecond (10,000 when using 80MHz Clock)
+// Need to divide by 8, because this is the only option for CH32V103
+#define SYSTICK_ONE_MILLISECOND ((uint32_t)FUNCONF_SYSTEM_CORE_CLOCK / 8 / 1000)
+// Number of ticks elapsed per microsecond (10 when using 80MHz Clock)
+#define SYSTICK_ONE_MICROSECOND ((uint32_t)FUNCONF_SYSTEM_CORE_CLOCK / 8 / 1000000)
+
+// Simple macro functions to give a arduino-like functions to call
+// millis() reads the incremented systick variable
+// micros() reads the raw SysTick Count, and divides it by the number of 
+// ticks per microsecond ( WARN: Wraps every 90 seconds!)
+#define millis() (systick_millis)
+#define micros() (SysTick->CNT / SYSTICK_ONE_MICROSECOND)
+
+// Incremented in the SysTick IRQ - in this example once per millisecond
+volatile uint64_t systick_millis;
+
+/*
+* Initialises the SysTick to trigger an IRQ with auto-reload, using HCLK/1 as
+* its clock source
+*/
+void systick_init(void)
+{
+	// Disable Systick
+	SysTick->CTLR = 0;
+	
+	// Set the compare register to trigger once per millisecond
+	uint64_t cmp_tmp = SYSTICK_ONE_MILLISECOND - 1;
+	// In CH32V103 we can write systick regs only in 8bit manner
+	SysTick->CMP0 = (uint8_t)cmp_tmp;
+	SysTick->CMP1 = (uint8_t)(cmp_tmp >> 8);
+	SysTick->CMP2 = (uint8_t)(cmp_tmp >> 16);
+	SysTick->CMP3 = (uint8_t)(cmp_tmp >> 24);
+	SysTick->CMP4 = (uint8_t)(cmp_tmp >> 32);
+	SysTick->CMP5 = (uint8_t)(cmp_tmp >> 40);
+	SysTick->CMP6 = (uint8_t)(cmp_tmp >> 48);
+	SysTick->CMP7 = (uint8_t)(cmp_tmp >> 56);
+
+	// Reset the Count Register, and the global millis counter to 0
```

Comment:
not ideal to reset the counter, I would do it the same as you did it in the IRQ.